compiler/src/iree/compiler/Dialect/LinalgExt/IR/LinalgExtOps.cpp

```diff
@@ -353,6 +353,34 @@ LogicalResult MapScatterOp::verify() {
   return success();
 }
 
+void MapScatterOp::insertTransformationAtStart(
+    OpBuilder &builder,
+    function_ref<SmallVector<Value>(ArrayRef<BlockArgument>)>
+        transformationBuilder,
+    int64_t numSourceIndices) {
+  Block &transformBody = getTransformationRegion().front();
+  SmallVector<BlockArgument> oldSourceIndices(transformBody.getArguments());
+  SmallVector<Type> indexTypes(numSourceIndices, builder.getIndexType());
+  SmallVector<Location> locs(numSourceIndices, getLoc());
+  SmallVector<BlockArgument> newSourceIndices(
+      transformBody.addArguments(indexTypes, locs));
+  OpBuilder::InsertionGuard g(builder);
+  builder.setInsertionPointToStart(&transformBody);
+  SmallVector<Value> newSourceIndicesTransformed(
+      transformationBuilder(newSourceIndices));
+  assert(oldSourceIndices.size() == newSourceIndicesTransformed.size() &&
+         "expected transformation to produce the same number of Values as the "
+         "previous number of source indices.");
+  for (auto [oldIdx, newIdx] :
+       llvm::zip_equal(oldSourceIndices, newSourceIndicesTransformed)) {
+    SmallVector<OpOperand *> uses(llvm::make_pointer_range(oldIdx.getUses()));
+    for (OpOperand *use : uses) {
```

Comment:
Why not iterate `oldIdx.getUses()` directly?

```suggestion
    for (OpOperand *use : oldIdx.getUses()) {
```

Reply:
Not sure why I did that :p

Probably forgot to remove it after refactoring.

Reply:
Unresolving this because I needed to undo the change. Setting the use actually removes it from the use list and adds the new use to the list, so it will mess with the iterator when just using oldIdx.getUses() directly. I had forgotten why I was constructing the new list of pointers, but this was the reason.

Reply:
I see, thanks!